src/metatags/metatags-auto-suggest.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { GetObjectCommand } from '@aws-sdk/client-s3';
+import { GenvarClient } from '@adobe/spacecat-shared-gpt-client';
+import { isObject } from '@adobe/spacecat-shared-utils';
+
+const EXPIRY_IN_SECONDS = 25 * 60;
+
+/**
+ * Returns the pre-signed url for a AWS S3 object with a defined expiry.
+ * This url will be consumed by Genvar API to access the scraped content.
+ * Pre-signed URl format: https://{bucket}.s3.{region}.amazonaws.com/{object-path}?{query-params}
+ * @param s3Client
+ * @param log
+ * @param scrapedData
+ * @returns {Promise<string>} Presigned url
+ */
+async function getPresignedUrl(s3Client, log, scrapedData) {
+  try {
+    const command = new GetObjectCommand({
+      Bucket: process.env.S3_SCRAPER_BUCKET_NAME,
+      Key: scrapedData.s3key,
+    });
+    return await getSignedUrl(s3Client, command, {
+      expiresIn: EXPIRY_IN_SECONDS,
+    });
+  } catch (error) {
+    log.error(`Error generating presigned URL for ${scrapedData.s3key}:`, error);
+    return '';
+  }
+}
+
+export default async function metatagsAutoSuggest(allTags, context, site) {
+  const { s3Client, dataAccess, log } = context;
+  const {
+    detectedTags,
+    extractedTags,
+    healthyTags,
+  } = allTags;
+  const { Configuration } = dataAccess;
+  const configuration = await Configuration.findLatest();
+  if (!configuration.isHandlerEnabledForSite('meta-tags-auto-suggest', site)) {
+    log.info('Metatags auto-suggest is disabled for site');
+    return detectedTags;
+  }
+  log.info('Generating suggestions for Meta-tags using Genvar.');
```

Comment:
please consider removing or using `debug` log level for log messages without context - they will not end up on Coralogix. As we're growing we sometimes run into daily capacity limits of log ingestion, as such logs should be useful to the developer if sent to coralogix. kindly check all your log statments :)